pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -305,6 +305,21 @@ func (e *EndpointController) syncService(key string) {
 		e.queue.Add(key) // Retry
 		return
 	}
+	if len(pods.Items) == 0 {
```

Comment:
Can you do https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replication/replication_controller.go#L500 instead?